### DISCUSSION TITLE
The pipe forwards the client host.

### DISCUSSION THREAD

**Discussion by royalpinto** [sindresorhus/got#1748] [Q&A] [answered]
#### Describe the bug

 - Node.js version: v14
 - OS & version: macOS (10)
 - Got version: 11.8.2

#### Actual behavior

The pipe request forwards all the client headers including the "Host"

#### Expected behavior

The pipe request should forwards all the client headers but not the "Host"

#### Code to reproduce

```bash
npm install got nock express request 
```

```js
const got = require('got')
const nock = require('nock')
const express = require('express')
const request = require('request')

nock('http://www.google.com')
  .persist()
  .get('/echo')
  .reply(function (uri, body) {
    return [200, { headers: this.req.headers }]
  })

const proxy = express()
proxy.get('/got/pipe', (req, res) => {
  req.pipe(got.stream('http://www.google.com/echo')).pipe(res)
})

proxy.get('/request/pipe', (req, res) => {
  req.pipe(request.get('http://www.google.com/echo')).pipe(res)
})

proxy.listen(3000, () => {
  console.log('Try the following:')
  console.log("curl -H 'Hi: Bye' -H 'Host: Value' 'http://127.0.0.1:3000/got/pipe'")
  console.log("curl -H 'Hi: Bye' -H 'Host: Value' 'http://127.0.0.1:3000/request/pipe'")
})
```

Output
```sh
// With Got
curl -H 'Hi: Bye' -H 'Host: Value' 'http://127.0.0.1:3000/got/pipe'
{
    "headers": {
        "accept": "*/*",
        "accept-encoding": "gzip, deflate, br",
        "hi": "Bye",
        "host": "Value", **// Should not have been this.**
        "user-agent": "got (https://github.com/sindresorhus/got)"
    }
}

// With Request
curl -H 'Hi: Bye' -H 'Host: Value' 'http://127.0.0.1:3000/request/pipe'
{
    "headers": {
        "accept": "*/*",
        "hi": "Bye",
        "host": "www.google.com",
        "user-agent": "curl/7.64.1"
    }
}
```

Since the host doesn't match the hostname in the URL, it leads to a certificate error when hitting the actual service than the mocked one.

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment, marked as answer]
Simply `delete req.headers.host;` before piping to got.

https://github.com/sindresorhus/got/blob/f896aa52abc41fe40d4942da94a0408477358f14/source/core/index.ts#L1415-L1422

Or you can use a `beforeRequest` hook:

```js
{
    hooks: {
        beforeRequest: [
            options => {
                delete options.headers.host;
            }
        ]
    }
  }
}
```

**Comment by royalpinto** [comment]
Hi @szmarczak ,

Is it not supposed to be handled in the library itself as `request` does?

**Reply by szmarczak** [reply]
> Is it not supposed to be handled in the library itself as request does?

Why would it be?

**Reply by royalpinto** [reply]
@szmarczak I am not sure, but I think in most of the cases, the hostname in the URL would be different for proxy and the target, and keeping the hostname the same will lead to certificate errors. I wonder in which cases the hostname needs to be preserved.

**Reply by szmarczak** [reply]
Well, I'm not actually sure if this is the most popular use case. We need more info.
